program-manager/src/program_update.rs

```diff
@@ -0,0 +1,303 @@
+                        .context(format!("Failed to find authorization {}", label))?;
+
+                    if let Some(not_before) = not_before {
+                        auth.not_before = not_before;
+                    }
+
+                    auth.priority = priority;
+                    auth.max_concurrent_executions = max_concurrent_executions;
+                }
+                AuthorizationInfoUpdate::Disable(label) => {
+                    verify_authorization_exists(&config.authorizations, label.clone())?;
+
+                    // Create instruction for disabling authorization
+                    instructions.push_back(WasmMsg::Execute {
+                        contract_addr: config.authorization_data.authorization_addr.clone(),
+                        msg: to_json_binary(&valence_authorization_utils::msg::ExecuteMsg::PermissionedAction(
+                            valence_authorization_utils::msg::PermissionedMsg::DisableAuthorization { label }
+                        )).unwrap(),
+                        funds: vec![]
+                    }.into());
+                }
+                AuthorizationInfoUpdate::Enable(label) => {
+                    verify_authorization_exists(&config.authorizations, label.clone())?;
+
+                    // Create instruction for enabling authorization
+                    instructions.push_back(WasmMsg::Execute {
+                        contract_addr: config.authorization_data.authorization_addr.clone(),
+                        msg: to_json_binary(&valence_authorization_utils::msg::ExecuteMsg::PermissionedAction(
+                            valence_authorization_utils::msg::PermissionedMsg::EnableAuthorization { label }
+                        )).unwrap(),
+                        funds: vec![]
+                    }.into());
+                }
+            }
+        }
+
+        // Add all new authorizations in a single message to be executed on the authorization contract
+        // We add it first because we want to update and add new authorizations before calling them
+        instructions.push_front(
+            WasmMsg::Execute {
+                contract_addr: config.authorization_data.authorization_addr.clone(),
+                msg: to_json_binary(
+                    &valence_authorization_utils::msg::ExecuteMsg::PermissionedAction(
+                        valence_authorization_utils::msg::PermissionedMsg::CreateAuthorizations {
+                            authorizations: new_authorizations,
+                        },
+                    ),
+                )
+                .unwrap(),
+                funds: vec![],
+            }
+            .into(),
+        );
+
+        // Save the updated config to the registry
+        neutron_connector.update_program_config(config).await?;
+
+        Ok(UpdateResponse {
+            instructions: instructions.into(),
+        })
+    }
+}
+
+fn verify_authorization_not_exists(
```

Comment:
This feels redundant ..

Reply:
Yea, but explicit in what we verify, and we want to error here, so I felt like it will be better encapsulated in a helper function instead of bool and errors in code it self.
Should I change it?